### DISCUSSION TITLE
Error -4

### DISCUSSION THREAD

**Discussion by smilejsu82** [naverz/zepeto-studio-kor#526] [Issue Report (Legacy)]
![image](https://user-images.githubusercontent.com/50188795/159841640-91e07abb-6698-4bb4-b511-7d308a100db6.png)

Error -4 이것은 무엇을 의미 하는것이죠?

다음은 마우스 클릭에 대한 테스트 입니다 

프로젝트를 빌드하고 업로드 하는중 에러가 발생했습니다 

이것이 무엇을 의미 하는지 모르겠습니다 

![image](https://user-images.githubusercontent.com/50188795/159841713-b9e131f8-5dd5-408d-9a9e-29447831b9aa.png)

https://user-images.githubusercontent.com/50188795/159844275-9fe69fe0-2b4d-4d45-93ae-5d12801d7943.mp4

**Comment by Zepeto-TechEvangelist** [comment]
안녕하세요. Upload시에 오류가 발생했다면 월드 콘솔 id 등록을 제대로 했는지 확인이 필요해 보입니다.
https://studio.zepeto.me/kr/guides/preparing-for-world-creating
위 가이드에 있는 내용이 잘 반영되었는지 확인 후 말씀주세요~

**Reply by smilejsu82** [reply]
월드 만들기를 통해 테스트용 월드를 생성했습니다 
![image](https://user-images.githubusercontent.com/50188795/159859851-780f63f0-7621-47a7-be1c-4efebdc21c8e.png)

카테고리및 스크린샷도 넣었습니다 
![image](https://user-images.githubusercontent.com/50188795/159859923-efe95f69-7bee-4955-9022-9b400ddbc697.png)

빌드세팅에서 package name도 확인 했습니다
![image](https://user-images.githubusercontent.com/50188795/159860036-e354f17f-65b2-4782-8bfd-e876ba2dd1e1.png)
 
![image](https://user-images.githubusercontent.com/50188795/159860019-50ca0f50-c431-4648-8bd2-efe68c604492.png)

로그인 확인도 했습니다 
![image](https://user-images.githubusercontent.com/50188795/159860084-68f25ee4-6fe1-4b45-ae8f-9f5d4e90170e.png)

제페토 월드 ID도 확인 했습니다 
![image](https://user-images.githubusercontent.com/50188795/159860130-71091586-50bd-4475-b7f3-63d480e1a91f.png)

여전히 에러가 납니다 
이후 어떻게 대처 해야 할까요?
![image](https://user-images.githubusercontent.com/50188795/159860266-1a8fc73e-ca27-4a07-a531-969a217489e5.png)
![image](https://user-images.githubusercontent.com/50188795/159860294-f17d9d32-0a34-4640-88b1-80e1739426dc.png)
![image](https://user-images.githubusercontent.com/50188795/159860324-cff088fc-8eb9-445f-812c-8918b15b1869.png)
![image](https://user-images.githubusercontent.com/50188795/159860433-cf21c489-4595-4fa7-a9e6-f12f223d4b06.png)


혹시 이것과 상관이 있나요?
이것은 정확히 어떤 기능을 하는지 알고 싶네요
![image](https://user-images.githubusercontent.com/50188795/159861049-0f6b021a-a33c-4041-9f98-815032b3c62e.png)

**Reply by Zepeto-TechEvangelist** [reply]
넵 정상적으로 월드 콘솔 등록은 완료하셨습니다. 오류 확인 해보겠습니다.

**Reply by smilejsu82** [reply]
답변 감사합니다 

혹시 이것에 대해서도 설명해주실수 있을까요 
![image](https://user-images.githubusercontent.com/50188795/159864896-4ad8f0c9-0b4b-4b34-8910-f311bb0eaeaa.png)

**Reply by Zepeto-TechEvangelist** [reply]
new 인풋 시스템을 활성화 할 것이냐에 대한 질문이므로 YES를 눌러주시면 됩니다.

**Reply by smilejsu82** [reply]
이것을 활성화 했을경우 OnMouseXXX 이벤트 함수가 호출되지 않습니다 
NO를 선택했을땐 정상 동작 하네요

**Reply by smilejsu82** [reply]
![image](https://user-images.githubusercontent.com/50188795/159865768-a2b142d9-299b-4dd0-99b7-1462403e0ae9.png)

생성된 package를 직접 업로드 하려 했으나 실패 했습니다 
![image](https://user-images.githubusercontent.com/50188795/159865945-887ebd0e-650c-4708-839a-42b7c08f231b.png)

**Reply by Zepeto-TechEvangelist** [reply]
네, 로그 참고하겠습니다.
프로젝트에 제페토 플레이어를 추가하고 빌드하면 잘 되는지도 한번 체크해 주세요. 
https://studio.zepeto.me/kr/guides/character-control

**Reply by smilejsu82** [reply]
![image](https://user-images.githubusercontent.com/50188795/159868682-c229e8b7-3999-4ad9-80f7-a809a6a93887.png)
제페토 플레이어를 붙이고 에디터에서 플레이 해본결과 정상 동작 합니다 

여전히 OnMouseDown은 동작 안하네요 
![image](https://user-images.githubusercontent.com/50188795/159868802-6f04a914-2e53-4734-98a2-a799953410b9.png)

퍼블리시 버튼을 눌렀을때 에러는 여전 합니다 
![image](https://user-images.githubusercontent.com/50188795/159868944-4ec274be-3061-44dc-abdb-12af9dbbf1c4.png)


System.IO.FileNotFoundException: Could not find file "C:\Users\smile\workspace\unity\ZEPETO_OnMouseDown\Build\ZEPETO.Script\windows"
File name: 'C:\Users\smile\workspace\unity\ZEPETO_OnMouseDown\Build\ZEPETO.Script\windows'
  at System.IO.FileStream..ctor (System.String path, System.IO.FileMode mode, System.IO.FileAccess access, System.IO.FileShare share, System.Int32 bufferSize, System.Boolean anonymous, System.IO.FileOptions options) [0x0019e] in <695d1cc93cca45069c528c15c9fdd749>:0 
  at System.IO.FileStream..ctor (System.String path, System.IO.FileMode mode, System.IO.FileAccess access, System.IO.FileShare share) [0x00000] in <695d1cc93cca45069c528c15c9fdd749>:0 
  at (wrapper remoting-invoke-with-check) System.IO.FileStream..ctor(string,System.IO.FileMode,System.IO.FileAccess,System.IO.FileShare)
  at System.IO.File.OpenRead (System.String path) [0x00000] in <695d1cc93cca45069c528c15c9fdd749>:0 
  at System.IO.File.ReadAllBytes (System.String path) [0x00000] in <695d1cc93cca45069c528c15c9fdd749>:0 
  at ZEPETO.Script.ZepetoScriptPackageFile.Save (System.IO.Stream stream) [0x000fa] in <11c5dafbc72a4c259f3b10ebd060003f>:0 
  at ZEPETO.World.Editor.ZepetoScriptPackageTask.Run () [0x00379] in <11c5dafbc72a4c259f3b10ebd060003f>:0 
  at ZEPETO.Script.Build.ZepetoScriptBuildContext.Build () [0x000df] in <d4cd6ac267b44c4d8103a88a7c72040b>:0 
UnityEngine.Debug:LogError (object

**Reply by smilejsu82** [reply]
새프로젝트를 만들어 실행해 봤습니다 
![image](https://user-images.githubusercontent.com/50188795/159875061-5f4cbe55-e5bc-44a1-b7ea-24ff0deb470b.png)

이번에는 Error -2 네요 

![image](https://user-images.githubusercontent.com/50188795/159874297-71e14da9-0db4-4693-b4bf-5d62c00c4d47.png)

제페토 플레이어를 붙이고 다시 테스트 해봤습니다 

새 프로젝트에서도 Error -4 는 여전 합니다 
![image](https://user-images.githubusercontent.com/50188795/159874984-cdc58f2b-8f9b-41ee-8b39-b005af338546.png)

![image](https://user-images.githubusercontent.com/50188795/159875809-0ba10d12-07d4-4edc-b30f-7624f004d44e.png)

**Reply by smilejsu82** [reply]
ZEPETO.world 버전을 1.1.5에서 1.1.2로 다운 그레이드했습니다 
첫번째 시도 후 에러 
![image](https://user-images.githubusercontent.com/50188795/159880819-fc661eeb-abad-43fd-b224-86d17d62a69a.png)
두번째 시도후 성공 
![image](https://user-images.githubusercontent.com/50188795/159880878-b8f68c20-249a-4c96-9efe-2186e9c9747a.png)

패키지 파일 등록도 성공 했습니다 
![image](https://user-images.githubusercontent.com/50188795/159881100-11d3dbcc-5dd1-4b4a-be9b-3350a28b5716.png)

에뮬에서도 실행이 성공적이였습니다 
![image](https://user-images.githubusercontent.com/50188795/159881454-48b0fda6-c70f-45fb-991c-51c85c5a81af.png)

모바일 디바이스(ios)에서도 잘 동작 하는것을 확인 했습니다 

**이것이 버전 문제인가요?**

**Reply by Zepeto-TechEvangelist** [reply]
안녕하세요. 위 이슈에 대해 내부에서 확인을 해보니 
Publish 명령 전에 씬에 변경된 내용 저장이 되지 않아 번들을 굽는 과정중에 실패한 것이 이유입니다.
Error -4 의 의미는 UnsavedChanges 입니다.

Publish 명령 전에 씬에 저장된 내용이 있으면 저장을 반드시 해주시기를 부탁 드립니다.

**Reply by smilejsu82** [reply]
답변 감사합니다 

하지만 publish이후 upload할때 문제점은 여전 합니다 
![image](https://user-images.githubusercontent.com/50188795/160088767-2939e2eb-bc00-4ad2-9762-146a93e2c5f9.png)
![image](https://user-images.githubusercontent.com/50188795/160088809-eb4166c8-6184-4e92-b36c-aa78810c224e.png)

일단 수동으로 패키지 파일을 업로드해서 사용하고 있습니다 
![image](https://user-images.githubusercontent.com/50188795/160089133-ff561126-74da-46c5-9385-e0b450fb2b45.png)

**Reply by Zepeto-TechEvangelist** [reply]
Error -4 는 해결이네요.
다른 프로젝트도 업로드 해보시고 같은 오류가 발생할 경우 별도의 게시물로 작성 부탁 드립니다.

**Reply by smilejsu82** [reply]
빠른 답변 감사합니다 ~ ^^
